tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/package.scala

```diff
@@ -1,15 +1,70 @@
 package at.forsyte.apalache.tla
 
-import at.forsyte.apalache.tla.lir.oper.TlaOper
 import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper.TlaOper
+import scalaz._
 
 import scala.language.implicitConversions
-import scalaz._
 
-// Contains classes, typedefs and implicit conversion methods used in the package
+/**
+ * The key definitions related to the typed builder. The most important ones for the users of this API are the methods:
+ *
+ *   - an implicit conversion `build` that converts a builder state to its final form, e.g., to `TlaEx`.
+ *   - an implicit conversion `liftBuildToSeq` that applies `build` to a sequence.
+ *
+ * To use the above methods in your code, import the implicit conversions as follows:
+ *
+ * {{{
+ *  import at.forsyte.apalache.tla.typecomp._
+ * }}}
+ */
 package object typecomp {
 
+  /**
+   * Build a data structure (e.g., `TlaEx` or `TlaOperDecl`), given a state of the builder.
+   *
+   * @param builderState
+   *   the internal state of the builder, which captures a data structure made so far
+   * @tparam T
+   *   the type of a data structure to build, e.g., `TlaEx` or `TlaOperDecl`.
+   * @return
+   *   the constructed data structure of type `T`
+   * @throws TBuilderTypeException
+   *   when a constructed expression is ill-typed
+   * @throws TBuilderScopeException
+   *   when a constructed expression has an incorrect scope
+   */
+  implicit def build[T](builderState: TBuilderInternalState[T]): T = builderState.run(TBuilderContext.empty)._2
+
+  /**
+   * An implicit conversion via a class that works as [[build]], but via a method call to `.build()`.
+   *
+   * @param builderState
+   *   the internal state of the builder, which captures a data structure made so far
+   * @tparam T
+   *   the type of a data structure to build, e.g., `TlaEx` or `TlaOperDecl`.
+   */
+  implicit class BuildViaMethod[T](builderState: TBuilderInternalState[T]) {
+    def build: T = {
+      builderState.run(TBuilderContext.empty)._2
+    }
```

Comment:
> Just call the implicit explicitly. Or even

I think I tried the first version, and Scala did not like it. The second one requires an implicit conversion, right? Are you sure it would work with the restriction of at most one implicit conversion?

Reply:
The second way works, to my surprise :)